### PR TITLE
Feature. DASH-480 Redesign FCH selection page

### PR DESCRIPTION
**Pull request by trukhilio**
- Create new FCH selection page
- Common UI fixes, additions
- CartItem Refactor (start)

### CODE REVIEW

File: client/src/components/AddressDomainCart/AddressDomainCart.tsx [hunk 1]
```diff
@@ -29,6 +29,7 @@ import {
   Prices,
   RedirectLinkData,
 } from '../../types';
+import { DOMAIN_TYPE } from '../../constants/fio';
```
Comment (vmkarpenko) [inline client/src/components/AddressDomainCart/AddressDomainCart.tsx:32]: sort order

File: client/src/components/AddressWidget/AddressWidget.tsx [hunk 2]
```diff
@@ -54,15 +51,7 @@ const AddressWidget: React.FC<Props> = props => {
...
-      <div className={classes.form}>
-        <AddressDomainForm
-          isHomepage={true}
-          type={ADDRESS}
-          isReverseColors={isReverseColors}
-          isDarkWhite={isDarkWhite}
-          {...props}
-        />
-      </div>
+      <FormComponent isReverseColors={isReverseColors} links={links} />
```
Comment (vmkarpenko) [inline client/src/components/AddressWidget/AddressWidget.tsx:54]: `isDarkWhite={isDarkWhite}
`missing

**Review by orkhan-huseyn — APPROVED** [review]

File: client/src/components/SelectionItem/components/DomainTypeBadge/DomainTypeBadge.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+import CommonBadge from '../../../Badges/CommonBadge/CommonBadge';
+import Modal from '../../../Modal/Modal';
+
+import { DOMAIN_TYPE_PARAMS } from '../../../../constants/fio';
+
+import { DomainItemType } from '../../../../types';
+
+import classes from './DomainTypeBadge.module.scss';
+import SubmitButton from '../../../common/SubmitButton/SubmitButton';
```
Comment (vmkarpenko) [inline client/src/components/SelectionItem/components/DomainTypeBadge/DomainTypeBadge.tsx:11]: sort order

**Review by vmkarpenko — APPROVED** [review]

File: client/src/components/SelectionItem/components/DomainTypeBadge/DomainTypeBadge.tsx (hunk 3 above)
Comment (orkhan-huseyn) [reply client/src/components/SelectionItem/components/DomainTypeBadge/DomainTypeBadge.tsx:11]: that's VSCode auto import :smiling_face_with_tear:

File: client/src/components/AddressWidget/AddressWidget.tsx (hunk 2 above)
Comment (trukhilio) [reply client/src/components/AddressWidget/AddressWidget.tsx:54]: Fixed

File: client/src/components/SelectionItem/components/DomainTypeBadge/DomainTypeBadge.tsx (hunk 3 above)
Comment (trukhilio) [reply client/src/components/SelectionItem/components/DomainTypeBadge/DomainTypeBadge.tsx:11]: Fixed

**Pull request merged**